### PR TITLE
IOS-15408: (part 5) allow cleaning up of background sessions if it's …

### PR DESCRIPTION
**Pull request by thuyBox**
…not the current background session and has no more pending tasks

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.h [hunk 1]
```diff
@@ -250,6 +250,19 @@ totalBytesExpectedToSend:(int64_t)totalBytesExpectedToSend;
...
+ * Clean up background session from cache and invalidate it if possible,
+ * i.e. if the background session to be cleaned up is not the current
+ * background session and has no pending tasks
+ *
+ * @param backgroundSessionID   background session ID to clean up
+ * @param error error cleaning up this background session
+ *
+ * @return YES if successfully clean up, NO otherwise
+*/
+- (BOOL)cleanUpBackgroundSessionIfPossibleGivenID:(NSString *)backgroundSessionID
```
Comment (roliva-box) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.h:263]: Nit: consider renaming this to simply `cleanUpBackgroundSession:`
Given that the result is a BOOL, for success/failure, the "if possible" is somewhat redundant.

**Review by roliva-box — APPROVED** [review]
LGTM with one nit comment.

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m [hunk 2]
```diff
@@ -562,6 +549,27 @@ - (BOOL)cleanUpBackgroundSessionTaskIfExistForUserId:(NSString *)userId associat
     return success;
 }
 
+- (BOOL)cleanUpBackgroundSessionIfPossibleGivenID:(NSString *)backgroundSessionID
+                                            error:(NSError **)error
+{
+    BOOL success = NO;
+    if (backgroundSessionID != nil && [backgroundSessionID isEqualToString:[self backgroundSessionIdentifier]] == NO) {
+        //this background session is not self.backgroundSession, most likely so we are running inside the main app
+        //and this background session is from extension, clean it up if it has no more pending tasks
```
Comment (jlawton) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m]: The above comment doesn't read correctly, I think.

**Review by jlawton — APPROVED** [review]

**Pull request merged**